jax/interpreters/xla.py

```diff
@@ -224,6 +226,14 @@ def apply_primitive(prim, *args, **params):
   compiled_fun = xla_primitive_callable(prim, *unsafe_map(arg_spec, args), **params)
   return compiled_fun(*args)
 
+
+def _partition_outputs(avals, outs):
+  nouts = [aval._num_buffers for aval in avals]
+  assert sum(nouts) == len(outs), f"Internal error: sum(nouts)={sum(nouts)} should equal len(outs)={len(outs)}."
```

Comment:
How about putting this and the next assertion under `if not core.skip_checks`?

Reply:
Done.